### PR TITLE
Add service restart for bootstrap node on first deployment

### PR DESCRIPTION
**Pull request by slam1987**
**Summary**

Currently the `mysql` process on the bootstrap node cannot be stopped by the `/etc/init.d/mysql` service.

The process ID will have to be manually terminated for `/etc/init.d/mysql` service to control the `mysql` process.

A workaround to this problem is to terminate the `mysql` process on the bootstrap node after initial deployment and rejoin it to the cluster using the `/etc/init.d/mysql` service.

This allows all nodes to operate in the same way therefore allowing rolling restarts throughout the cluster.

Refer to reported bug: [https://bugs.launchpad.net/ubuntu/+source/percona-xtradb-cluster-5.6/+bug/1583000](https://bugs.launchpad.net/ubuntu/+source/percona-xtradb-cluster-5.6/+bug/1583000)

**Tests**

Tests were done using `serial: 1` in `setup_keycloak.yml` when calling the role.

Initial xtradb deployment:
( Confirm bootstrap node is skipped when run first )
```
TASK [ansible-role-xtradb-cluster : Checking if bootstrap node has rejoined the cluster] ******************************
skipping: [keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Terminate mysql process on bootstrap node] ****************************************
skipping: [keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Wait for mysql process to stop on bootstrap node] *********************************
skipping: [keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Start mysql service on bootstrap node] ********************************************
skipping: [keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Marking boostrap node rejoin is changed] ******************************************
skipping: [keycloak-db-controller001-dev.core-x.net]
```

Initial xtradb deployment:
( Confirm last node in cluster delegates to bootstrap node )
```
TASK [ansible-role-xtradb-cluster : Checking if bootstrap node has rejoined the cluster] ******************************
ok: [keycloak-db-controller002-dev.core-x.net -> keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Terminate mysql process on bootstrap node] ****************************************
changed: [keycloak-db-controller002-dev.core-x.net -> keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Wait for mysql process to stop on bootstrap node] *********************************
ok: [keycloak-db-controller002-dev.core-x.net -> keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Start mysql service on bootstrap node] ********************************************
changed: [keycloak-db-controller002-dev.core-x.net -> keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Marking boostrap node rejoin is changed] ******************************************
changed: [keycloak-db-controller002-dev.core-x.net -> keycloak-db-controller001-dev.core-x.net]
```

Subsequent playbook run after initial deployment:
```
TASK [ansible-role-xtradb-cluster : Checking if bootstrap node has rejoined the cluster] ******************************
ok: [keycloak-db-controller002-dev.core-x.net -> keycloak-db-controller001-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Terminate mysql process on bootstrap node] ****************************************
skipping: [keycloak-db-controller002-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Wait for mysql process to stop on bootstrap node] *********************************
skipping: [keycloak-db-controller002-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Start mysql service on bootstrap node] ********************************************
skipping: [keycloak-db-controller002-dev.core-x.net]

TASK [ansible-role-xtradb-cluster : Marking boostrap node rejoin is changed] ******************************************
skipping: [keycloak-db-controller002-dev.core-x.net]
```

Confirm that bootstrap node can be stopped and started using `systemctl`:
```
stanleylam@keycloak_db-controller001-dev:~$ sudo systemctl stop mysql
stanleylam@keycloak_db-controller001-dev:~$ ps aux | grep mysql
stanley+  7260  0.0  0.0  14856  1064 pts/0    S+   21:54   0:00 grep --color=auto mysql
stanleylam@keycloak_db-controller001-dev:~$ sudo systemctl start mysql
stanleylam@keycloak_db-controller001-dev:~$ sudo cat /var/run/mysqld/mysqld.pid
7906
```

Confirm that bootstrap node can successfully rejoined the cluster after restart:
( excerpt from `/var/log/mysqld.log` )
```
2020-02-11T22:48:24.273325Z 0 [Note] WSREP: Quorum results:
	version    = 6,
	component  = PRIMARY,
	conf_id    = 13,
	members    = 2/2 (primary/total),
	act_id     = 18,
	last_appl. = -1,
	protocols  = 0/9/3 (gcs/repl/appl),
	group UUID = 8d097d49-4d1e-11ea-a114-a68004f06614
```

### CODE REVIEW

File: tasks/postinstall.yml [hunk 1]
```diff
@@ -85,3 +85,32 @@
       audit_log_include_accounts="{{ xtradb_auditlog_account }}@%"
   when: xtradb_auditlog_enabled
   notify: restart xtradb
+
+- name: Terminate mysql process on bootstrap node
```
Comment (nikosmeds) [inline tasks/postinstall.yml]: We could play with this?

```
delegate_to: "{{ groups[xtradb_nodes_group | first }}"
```

and

```
when: "{{ groups[xtradb_nodes_group] | last }}"
```

Comment (slam1987) [reply tasks/postinstall.yml]: Done.

**Comment by wified** [comment]
My opinion is that we can simplify this drastically. The reason it will not stop when we issue `systemctl stop mysql` is because it was not started with that so systemd does not know about it. If you look in the playbook you will see this:
```
- name: Start the master node
  command: /etc/init.d/mysql bootstrap-pxc
  when:
    - xtradb_bind_address == xtradb_master_node
    - ansible_os_family == "Debian"
```
What if we just create a systemd service file for it something like `/etc/systemd/system/mysql.bootstrap.service` and then issue a stop in the postinstall.yml?

**Comment by wified** [comment]
Also on a side note i am not seeing this happen anymore in dev?

**Comment by wified** [comment]
After discussing all good

**Pull request merged**